components/ToggleButton.js

```diff
@@ -2,23 +2,48 @@ import { makeStyles } from '@material-ui/core/styles';
 import Button from '@material-ui/core/Button';
 import clsx from 'clsx';
 import Grid from '@material-ui/core/Grid';
+import Popover from '@material-ui/core/Popover';
 import PropTypes from 'prop-types';
 import React, { useState, useEffect } from 'react';
+import Typography from '@material-ui/core/Typography';
 
 const useStyles = makeStyles(theme => ({
   button: {
     margin: theme.spacing(1, 2, 1, 0),
   },
+  popover: {
+    pointerEvents: 'none',
+  },
+  popoverPaper: {
+    padding: theme.spacing(1),
+    backgroundColor: 'black',
+    color: 'white',
+    boxShadow: 'none',
+  },
 }));
 
 function ToggleButton(props) {
-  const { options, value, handleChange, multiSelect, buttonClassName, buttonVariant } = props;
-  const [selected, setSelected] = useState(value);
   const classes = useStyles();
+  const {
+    options,
+    disabledOptions,
+    value,
+    handleChange,
+    multiSelect,
+    buttonClassName,
+    buttonVariant,
+    showPopover,
+    disabledMessage,
+  } = props;
+  const [selected, setSelected] = useState(value);
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [openPopover, setOpenPopover] = useState(false);
 
+  const isDisabled = v => disabledOptions.includes(v);
   const isSelected = v => (multiSelect ? selected.includes(v) : selected === v);
   const addSelected = v => (multiSelect ? [...selected, v] : v);
   const removeSelected = v => (multiSelect ? selected.filter(el => el !== v) : '');
+  const hasPopover = v => showPopover && isDisabled(v);
```

Comment:
Seems a little odd to have a `showPopover` prop -- makes it seem like if true, then a popover will be open.

Beyond the naming of it, it seems that we don't really need this prop -- seems like we wouldn't ever pass in a popover message without wanting to show it.

Could we nix the `showPopover` prop entirely and make this line:

```JS
const hasPopover = v => disabledMessage && isDisabled(v);
```

Reply:
Agree, the prop isn't necessary and the naming is confusing. removed.